### ISSUE TITLE
bootstrap raises and error

### ISSUE THREAD

**Issue by spedygiorgio** [spedygiorgio/markovchain#15]
data(rain)
sequs<-rain$rain
mcBoot<-markovchainFit(data = sequs,nboot = 10,method="bootstrap") #ok
mcBoot2<-markovchainFit(data = sequs,nboot = 200,method="bootstrap") #ok but slower
mcBoot<-markovchainFit(data = sequs[1:100],nboot = 100,method="bootstrap",parallel=TRUE) #parallel. Error: Error: not compatible with requested type

**Comment by spedygiorgio** [comment]
Also, please, put in the vignette, an example on using the parallel feature for bootstrap. Thanks in advance.

**Comment by spedygiorgio** [comment]
Ok, task done. When you have time put the standard error after the estimate slot: estimate, standard error, confidence intervals and simulation...

**Comment by tskang** [comment]
Removed the error, but there are still problems. With small nboot values like 10, it works fine. However, errors occur with large nboot values. I'll try to fix this soon.

markovchainFit(data = sequs[1:100],nboot = 10,method="bootstrap",parallel=TRUE) # ok
markovchainFit(data = sequs[1:100],nboot = 20,method="bootstrap",parallel=TRUE) # ok or warning
markovchainFit(data = sequs[1:100],nboot = 30,method="bootstrap",parallel=TRUE) # Error: 'dimnames' applied to non-array
markovchainFit(data = sequs[1:100],nboot = 50,method="bootstrap",parallel=TRUE) # Error
markovchainFit(data = sequs[1:100],nboot = 100,method="bootstrap",parallel=TRUE) # Error in setwd(old) : character argument expected

**Comment by spedygiorgio** [comment]
probably a memory issue...

**Comment by tskang** [comment]
Added an example to the vignette.

**Comment by tskang** [comment]
Removed the error. 
Had to modify the parallel worker as RcppParallel does not support complex data types such as List (in our case, List of CharacterVector). It supports only Matrix or Vector type over which we can iterate with ParallelFor because of thread safety issue. Instead of parallelizing over the list in _mcFitBootStrap function, I parallelized over each CharacterVector in the createSequenceMatrix function.
